lib/HistoricSync.js

```diff
@@ -28,12 +28,48 @@ function HistoricSync(opts) {
   opts = opts || {};
   this.shouldBroadcast = opts.shouldBroadcastSync;
 
+  switch (config.network) {
+    case 'regtest':
+      var hex = function (hex) {
+        return new Buffer(hex, 'hex');
+      };
+      this.network = {
+        name: 'regtest',
+        magic: hex('fabfb5da'),
+        addressVersion: 0x6f,
+        privKeyVersion: 239,
+        P2SHVersion: 196,
+        hkeyPublicVersion: 0x043587cf,
+        hkeyPrivateVersion: 0x04358394,
+        genesisBlock: {
+          hash: hex('0f9188f13cb7b2c71f2a335e3a4fc328bf5beb436012afca590b1a11466e2206'),
+          merkle_root: hex('4a5e1e4baab89f3a32518a88c31bc87f618f76673e2cc77ab2127b7afdeda33b'),
+          height: 0,
+          nonce: 2,
+          version: 1,
+          prev_hash: buffertools.fill(new Buffer(32), 0),
+          timestamp: 1296688602,
+          bits: 545259519,
+        },
+        dnsSeeds: [],
+        defaultClientPort: 18444
+      };
+      networks['regtest'] = this.network;
+      break;
+    case 'testnet':
+      this.network = networks.testnet;
+      break;
+    default:
+      this.network = networks.livenet;
+  }
   this.network = config.network === 'testnet' ? networks.testnet: networks.livenet;
 
-  var genesisHashReversed = new Buffer(32);
-  this.network.genesisBlock.hash.copy(genesisHashReversed);
-  buffertools.reverse(genesisHashReversed);
-  this.genesis = genesisHashReversed.toString('hex');
+  // TODO: ?????  why reverse the bytes? why not just compare it to the string of the expected hash?
```

Comment:
This is a change to the existing code I wasn't happy to make, but I need some help understanding how it works and restating the hash of my regtest genesis block so it can work.